test/common/stats/symbol_table_fuzz_test.cc

```diff
@@ -24,6 +24,20 @@ DEFINE_FUZZER(const uint8_t* buf, size_t len) {
     // string before comparing.
     absl::string_view trimmed_fuzz_data = StringUtil::removeTrailingCharacters(next_data, '.');
     FUZZ_ASSERT(trimmed_fuzz_data == symbol_table.toString(stat_name));
+
+    // Also encode the string directly, without symbolizing it.
+    MemBlockBuilder<uint8_t> mem_block(SymbolTableImpl::Encoding::totalSizeBytes(next_data.size()));
+    SymbolTableImpl::Encoding::appendEncoding(next_data.size(), mem_block);
+    const uint8_t* data = reinterpret_cast<const uint8_t*>(next_data.data());
+    mem_block.appendData(absl::MakeSpan(data, data + next_data.size()));
+    FUZZ_ASSERT(mem_block.capacityRemaining() == 0);
+    absl::Span<uint8_t> span = mem_block.span();
+    std::pair<uint64_t, uint64_t> number_consumed =
+        SymbolTableImpl::Encoding::decodeNumber(span.data());
+    FUZZ_ASSERT(number_consumed.first == next_data.size());
```

Comment:
Would you be able to factor this out in to a utility test function and write a few unit tests for this functionality? (Or make use of `serializeDeserialize()` by making the block_size variable, and reuse the unit tests) We've had a lot of issues with bad logic in fuzz tests itself, so just want to have verification the code here is working as expected)

Reply:
done;ptal